internal/envoy/v3/listener.go

```diff
@@ -376,7 +376,7 @@ func (b *httpConnectionManagerBuilder) AddFilter(f *http.HttpFilter) *httpConnec
 		panic("Can't add more than one router to a filter chain")
 	}
 
-	if routerIndex != lastIndex {
+	if routerIndex != -1 && routerIndex != lastIndex {
```

Comment:
could make line 375 `if routerIndex != -1 {` and have that if wrap that one and this one to make it so its a little less redundant

Reply:
done